### ISSUE TITLE
Archive Slicer2 and Slicer3 releases

### ISSUE THREAD

**Issue by lassoan** [Slicer/Slicer#7004]
Slicer3 releases are available in this "special" location: https://www.slicer.org/slicer3-downloads/
It should be available at some standard location (maybe hosted on github?) and linked from download page and release notes.

A user asked about Slicer2 releases. If we still have binaries then it would be nice to archive them somewhere, too.

**Comment by pieper** [comment]
I agree.  In case we don't have them handy somewhere else, I have several versions of slicer2, 3, and 4, including a 2002 version from a demo CD we handed out at RSNA.  It still runs fine on a windows 11 laptop 20+ years later.  It would be nice to archive them somewhere.

![image](https://github.com/Slicer/Slicer/assets/126077/6ccecad3-97d7-4d90-9e13-f5ac846bac61)
